web/src/main/java/org/springframework/security/web/servlet/util/matcher/MvcRequestMatcher.java

```diff
@@ -95,6 +96,12 @@ private MatchableHandlerMapping getMapping(HttpServletRequest request) {
 
 	@Override
 	public MatchResult matcher(HttpServletRequest request) {
+		if (this.method != null && !this.method.name().equals(request.getMethod())) {
```

Comment:
Instead of duplicating the lines, would it be simpler to have `matches(request)` and `matcher(request)` call the same underlying private method?

Reply:
Done.